pkg/registry/resource/resourceslice/strategy.go

```diff
@@ -78,7 +79,10 @@ func (resourceSliceStrategy) PrepareForUpdate(ctx context.Context, obj, old runt
 }
 
 func (resourceSliceStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object) field.ErrorList {
-	return validation.ValidateResourceSliceUpdate(obj.(*resource.ResourceSlice), old.(*resource.ResourceSlice))
+	slice := obj.(*resource.ResourceSlice)
+	oldSlice := old.(*resource.ResourceSlice)
+	opts := validation.ValidationOptionsForResourceSlice(slice, oldSlice)
```

Comment:
@pohly @thockin this is a brand new API we're just promoting to beta in this release... did we know we were promoting to beta with a validation gap, and can we just close this unconditionally in beta?

Reply:
I had not looked into `ValidateNodeSelector` as it was already there and only became aware of these gaps when this PR came in.

I think we can unconditionally strengthen the validation for 1.32. I can see some small risk for resources created in 1.31 which are now invalid, but that shouldn't be a problem in practice.

Reply:
Thanks, I'd be supportive of that. Mind opening a separate issue / PR asap and tagging in @thockin, myself, and release folks?

Reply:
So you don't want to merge this PR for 1.32, just the subset of it which tightens validation for resource.k8s.io?

Reply:
I didn't catch up here. 
Are you suggesting we separate the changes under `pkg/apis/resource`, and merge that for 1.32 because the feature will be beta in 1.32?

Reply:
This PR already validated resource.k8s.io unconditionally, so merging it would solve the issue. But I can see how it might be too large to be included in 1.32.

I went ahead and stripped it down to just the parts which are relevant for DRA. It's strictly a subset of this PR with two TODOs where volume and pod validation calls the extended `ValidateNodeSelector`, so I kept @AxeZhan as author and only updated the commit message. Rebasing this PR on top of it shouldn't be hard.

=> https://github.com/kubernetes/kubernetes/pull/128932

Reply:
https://github.com/kubernetes/kubernetes/pull/128932 merged, @AxeZhan if you rebase on top of that, it'll cut this down to just the pod / pv changes, and we can queue that up for merge in 1.33

Reply:
If you squash first and then rebase one commit the conflicts should be minimal.